packages/maskbook/src/plugins/Trader/apis/bancor/index.ts

```diff
@@ -0,0 +1,76 @@
+import type {
+    SwapBancorRequest,
+    ExpectedTargetAmountResponse,
+    ExpectedSourceAmountResponse,
+    TradeTransactionCreationResponse,
+} from '../../types/bancor'
+import urlcat from 'urlcat'
+import { BANCOR_API_BASE_URL } from '../../constants'
+import { calculateMinimumReturn } from './calculateMinimumReturn'
+
+const getTargetAmount = async (request: SwapBancorRequest) => {
+    const baseUrl = BANCOR_API_BASE_URL[request.chainId!]
+    const url = urlcat(baseUrl, '/pricing/target-amount', {
+        source_dlt_type: 'ethereum',
+        source_dlt_id: request.fromToken?.address,
+        target_dlt_type: 'ethereum',
+        target_dlt_id: request.toToken?.address,
+        amount: Number(request.fromAmount!),
+    })
+    const response = await fetch(url)
+    if (response.ok) {
+        const data: ExpectedTargetAmountResponse = await response.json()
+        return data.amount
+    }
+    return undefined
+}
+
+const getSourceAmount = async (request: SwapBancorRequest) => {
+    const baseUrl = BANCOR_API_BASE_URL[request.chainId!]
+    const url = urlcat(baseUrl, '/pricing/source-amount', {
+        source_dlt_type: 'ethereum',
+        source_dlt_id: request.fromToken?.address,
+        target_dlt_type: 'ethereum',
+        target_dlt_id: request.toToken?.address,
+        amount: Number(request.toAmount!),
+    })
+    const response = await fetch(url)
+    if (response.ok) {
+        const data: ExpectedSourceAmountResponse = await response.json()
+        return data.amount
+    }
+    return undefined
+}
+
+export const swapTransactionBancor = async (request: SwapBancorRequest) => {
+    const baseUrl = BANCOR_API_BASE_URL[request.chainId]
+    const url = urlcat(baseUrl, '/transactions/swap', {
+        source_dlt_type: 'ethereum',
```

Comment:
Are multi-chains supported?

Reply:
Not yet, Bancor is currently focused on v3 version (which is not released yet), other chains support will be added once the v3 is done. 

API's were created recently and i think they are just done to support multi-chains in future.